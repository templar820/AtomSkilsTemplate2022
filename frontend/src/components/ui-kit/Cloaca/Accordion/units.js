import { css } from '@emotion/core';
import styled from '@emotion/styled';
import applyEmotionStyle from 'lib-root/utils/applyEmotionStyle';

const StyledPanelInner = styled.div`
    margin-bottom: 40px;
    padding-bottom: 1px;
    box-sizing: border-box;
    ${({ styles, ...restProps }) => applyEmotionStyle(styles, restProps)};
`;

const StyledPanelHead = styled.div`
    ${({ colors, expanded, disabled }) => css`
        position: relative;
        min-height: 72px;
        font-weight: 500;
        font-size: 15px;
        line-height: 1;
        color: ${colors.GrayScale_700};
        border-top: 1px solid ${colors.GrayScale_200};
        border-bottom: 1px solid ${colors.GrayScale_200};
        cursor: pointer;
        padding: 30px 24px 1px 24px;
        background-color: ${colors[expanded ? 'GrayScale_100' : 'GrayScale_0']};
        ${disabled &&
            css`
                opacity: 0.3;
                cursor: not-allowed;
            `};

        &:after {
            content: '';
            position: absolute;
            right: 20px;
            top: 30px;
            width: 9px;
            height: 9px;
            border-right: 2px solid ${colors.GrayScale_700};
            border-bottom: 2px solid ${colors.GrayScale_700};
            transform: rotate(${expanded ? '-135deg' : '45deg'});
            transition: 0.5s transform ease;
        }
    `};
    ${({ styles, ...restProps }) => applyEmotionStyle(styles, restProps)};
`;

const collapseContent = (contentHeight, expanded, animated) => {
    if (expanded === true) {
        return css`
            ${animated ? 'max-height: none;' : ' max-height: ' + contentHeight + 'px;'}
        `;
    } else {
        return css`
            ${animated ? 'max-height: ' + contentHeight + 'px;' : ' max-height: 0px;'}
        `;
    }
};

const StyledContentCollapse = styled.div`
    overflow: hidden;
    transition: 0.3s max-height ease;

    ${({ expanded, contentHeight, animated }) => collapseContent(contentHeight, expanded, animated)};
    ${({ styles, ...restProps }) => applyEmotionStyle(styles, restProps)};
`;

const StyledPanelContent = styled.div`
    overflow: hidden;
    padding: 93px 72px 10px 40px;
    ${({ styles, ...restProps }) => applyEmotionStyle(styles, restProps)};
`;

const LoaderComponentWrapper = styled.div`
    position: relative;
    height: 50px;
    ${({ styles, ...restProps }) => applyEmotionStyle(styles, restProps)};
`;

const EmptyStatesWrapper = styled.div`
    position: relative;
    min-height: 200px;
    ${({ styles, ...restProps }) => applyEmotionStyle(styles, restProps)};
`;

const StyledAccordionWrapper = styled.div`
    ${({ wrapperStyle, ...restProps }) => applyEmotionStyle(wrapperStyle, restProps)};
`;

export {
    StyledPanelInner,
    StyledPanelHead,
    StyledPanelContent,
    StyledContentCollapse,
    LoaderComponentWrapper,
    EmptyStatesWrapper,
    StyledAccordionWrapper
};
