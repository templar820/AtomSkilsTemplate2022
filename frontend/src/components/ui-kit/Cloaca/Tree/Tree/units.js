import React, { useState, useRef, useEffect } from 'react';
import styled from '@emotion/styled';
import { setClassName } from 'lib-root/utils/styleMixins';

const StyledWrapper = styled.div`
    position: relative;
    &:after {
        content: '';
        opacity: ${({ isFaderShown }) => (isFaderShown ? 1 : 0)};
        transition: 0.1s opacity;
        position: absolute;
        top: 0;
        bottom: 9px;
        right: 0;
        width: 100px;
        pointer-events: none;
        background: linear-gradient(to right, rgba(255, 255, 255, 0), rgba(255, 255, 255, 1));
    }
`;

const StyledSubWrapper = styled.div`
    padding-bottom: 9px;
    overflow-x: ${({ horizontalOverflow }) => horizontalOverflow};
`;

function checkFader(element, callback) {
    if (element) callback(element.scrollWidth - element.clientWidth - element.scrollLeft > 0);
}

function FadingWrapper({ children, expandedTree, animationDuration, horizontalOverflow }) {
    const [isFaderShown, setIsFaderShown] = useState(true);
    const subWrapperRef = useRef(null);

    useEffect(() => {
        setTimeout(function() {
            checkFader(subWrapperRef.current, setIsFaderShown);
        }, animationDuration);
    }, [expandedTree]);

    return (
        <StyledWrapper {...{ isFaderShown }}>
            <StyledSubWrapper
                {...{ horizontalOverflow, ref: subWrapperRef, children }}
                onScroll={() => checkFader(subWrapperRef.current, setIsFaderShown)}
            />
        </StyledWrapper>
    );
}

function Wrapper(props) {
    const { children, className, horizontalOverflow } = props;
    return horizontalOverflow ? (
        <FadingWrapper {...props} className={setClassName({ props: { className }, name: 'tree__fading-wrapper' })}>
            {children}
        </FadingWrapper>
    ) : (
        children
    );
}

export { Wrapper };
