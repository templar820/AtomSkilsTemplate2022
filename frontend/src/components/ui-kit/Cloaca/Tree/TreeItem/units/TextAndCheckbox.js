import React, { useContext, useMemo } from 'react';

import { ColorsContext } from 'lib-root/colors';

import Checkbox from 'lib-ui/Checkbox';
import { HighlightWrapper } from 'lib-ui/utils';
import {
    calcCheckboxStyle,
    calcCheckboxSquareStyle,
    calcCheckboxCaptionStyle,
    StyledCheckboxText,
    defaultHighlight
} from './units';

function TextAndCheckbox({
    occurrence,
    highlight,
    title: _title,
    isCheckable,
    checked,
    selection,
    handleChecking,
    id,
    checkboxStyle,
    checkboxSquareStyle,
    checkboxCaptionStyle,
    addTitlesAttrs
}) {
    const title = useMemo(() => {
        if (typeof _title === 'function') return _title({ occurrence });
        return _title;
    }, [_title, occurrence]);
    const colors = useContext(ColorsContext);
    const textWithHighlight = occurrence ? (
        <HighlightWrapper {...{ occurrence, highlight: defaultHighlight({ colors, highlight }) }}>
            {title}
        </HighlightWrapper>
    ) : (
        title
    );
    const browserTitle = addTitlesAttrs && typeof title === 'string' ? title : undefined;
    const styledText = (
        <StyledCheckboxText {...{ selection }} title={browserTitle}>
            {textWithHighlight}
        </StyledCheckboxText>
    );

    return !isCheckable ? (
        styledText
    ) : (
        <Checkbox
            onCheckChange={() => handleChecking({ [id]: checked !== true })}
            css={calcCheckboxStyle(checkboxStyle)}
            squareStyle={calcCheckboxSquareStyle(colors, checkboxSquareStyle, selection, checked)}
            captionStyle={calcCheckboxCaptionStyle(checkboxCaptionStyle)}
            checked={checked}>
            {styledText}
        </Checkbox>
    );
}

export default TextAndCheckbox;
