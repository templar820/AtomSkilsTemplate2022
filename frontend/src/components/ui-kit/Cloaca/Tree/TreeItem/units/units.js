import React, { useContext } from 'react';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { hexToRGBA } from 'lib-root/utils/styleMixins';
import { ColorsContext } from 'lib-root/colors';

import InlineIcon from 'lib-ui/InlineIcons';
import Spinner from 'lib-ui/Spinner';
import applyEmotionStyle from 'lib-root/utils/applyEmotionStyle';

// TopLine units
const StyledTopLine = styled.span`
    display: flex;
    align-items: flex-start;
    min-height: 24px;
    z-index: 2;
    ${({ selection, itemHeight }) =>
        selection &&
        css`
            height: ${itemHeight}px;
            align-items: center;
            position: relative;
        `}
    ${({ topLineStyle, ...restProps }) => applyEmotionStyle(topLineStyle, restProps)};

    ${({ nestingLevel, selection }) =>
        nestingLevel <= 1
            ? selection && 'padding-left:' + nestingLevel * 29 + 'px;'
            : selection && 'padding-left:' + nestingLevel * 36 + 'px;'}
`;

const StyledOverlay = styled.div`
    ${({ colors, selection, hovered, itemHeight }) =>
        selection &&
        css`
            cursor: pointer;
            content: '';
            position: absolute;
            right: 0;
            top: 0;
            width: 100vw;
            z-index: 1;
            height: ${itemHeight}px;
            background-color: ${hovered && hexToRGBA(colors.GrayScale_100, 0.5)};
        `};

    ${({ colors, checked, parentChecked, selection }) =>
        selection &&
        checked === true &&
        parentChecked !== true &&
        css`
            background-color: ${hexToRGBA(colors.GrayScale_200, 0.5)};
        `};
    ${({ overlayStyles, ...restProps }) => applyEmotionStyle(overlayStyles, restProps)};
`;

// TextAndCheckbox units
const calcCheckboxStyle = (extraStyles) => css`
    align-items: flex-start;
    width: 100%;
    ${applyEmotionStyle(extraStyles)}
`;

const calcCheckboxSquareStyle = (colors, extraStyles, selectionStatus, checkedStatus) => css`
    margin-top: 0;
    margin-bottom: 0;
    margin-left: 0;
    ${selectionStatus &&
        css`
            border: none;
            color: ${colors.GrayScale_700};
            order: 3;
            margin-left: 128px;
        `}
    ${selectionStatus &&
        checkedStatus === 'indeterminate' &&
        css`
            display: none;
        `}    
    ${applyEmotionStyle(extraStyles)}
`;

const calcCheckboxCaptionStyle = (extraStyles) => css`
    display: inline-block;
    vertical-align: middle;
    width: 100%;
    line-height: 20px;
    ${applyEmotionStyle(extraStyles)};
`;

const StyledCheckboxText = styled.span`
    display: inline-block;
    vertical-align: middle;
    overflow: hidden;
    text-overflow: ellipsis;
    width: 100%;
    ${({ selection }) =>
        selection &&
        css`
            white-space: nowrap;
        `}
`;

// Icon  units
const IconContainer = styled.span`
    min-width: 20px;
    min-height: 20px;
    margin-right: 14px;
    position: relative;
`;
const IconPositioner = styled.span`
    display: flex;
    align-items: center;
    justify-content: center;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 100%;
    height: 100%;
`;

const DotIcon = styled.span`
    display: block;
    width: 3.75px;
    height: 3.75px;
    border-radius: 50%;
    background-color: ${({ selection, colors }) => (selection ? colors.GrayScale_200 : colors.primary)};
    ${({ dotIconStyles, ...restProps }) => applyEmotionStyle(dotIconStyles, restProps)};
`;

const defaultLoadingStyle = css`
    display: inline-block;
    vertical-align: middle;
    width: 18.75px;
`;

const arrowStyle = (expanded, animationDuration) => css`
    transition: ${animationDuration + 'ms'} transform;
    cursor: pointer;
    ${!expanded &&
        css`
            transform: rotate(-90deg);
        `}
`;
const ArrowIcon = ({ colors, expanded, animationDuration }) => (
    <InlineIcon
        w="20px"
        h="20px"
        color={colors.GrayScale_200}
        css={arrowStyle(expanded, animationDuration)}
        icon="Arrow_type_1"
    />
);

const DefaultLoadingIcon = () => <Spinner wrapperStyle={defaultLoadingStyle} size={6} />;

const DefaultIcon = ({ expanded, hasSubtree, animationDuration, selection, dotIconStyles }) => {
    const colors = useContext(ColorsContext);
    return hasSubtree ? (
        <ArrowIcon {...{ colors, expanded, animationDuration }} />
    ) : (
        <DotIcon {...{ colors, selection, dotIconStyles }} />
    );
};

// TreeItem units

const defaultHighlight = ({ colors, highlight }) => css`
    color: ${colors.primary};
    ${highlight};
`;

const StyledTreeItem = styled.li`
    ${({ colors, disabled, isLined, selection, itemStyle, ...restProps }) => css`
        display: block;
        position: relative;
        padding: 3px 0;
        ${disabled &&
            css`
                opacity: 0.2;
                cursor: default;
                pointer-events: none;
                user-select: none;
            `};
        ${isLined &&
            css`
                position: relative;
                &:not(:last-of-type):before {
                    content: '';
                    position: absolute;
                    top: 23px;
                    left: 10px;
                    bottom: -3px;
                    width: 1px;
                    pointer-events: none;
                    background: ${colors.GrayScale_100};
                }
            `};
        ${selection &&
            css`
                padding: 0;
            `}
        ${applyEmotionStyle(itemStyle, { colors, disabled, isLined, selection, ...restProps })};
    `}
`;

export {
    StyledTopLine,
    calcCheckboxStyle,
    calcCheckboxSquareStyle,
    calcCheckboxCaptionStyle,
    StyledCheckboxText,
    IconContainer,
    IconPositioner,
    DefaultLoadingIcon,
    DefaultIcon,
    defaultHighlight,
    StyledTreeItem,
    StyledOverlay
};
