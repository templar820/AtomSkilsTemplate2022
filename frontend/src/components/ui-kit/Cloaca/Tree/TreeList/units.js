import { css } from '@emotion/core';
import styled from '@emotion/styled';
import applyEmotionStyle from 'lib-root/utils/applyEmotionStyle';

const StyledTreeList = styled.ul`
    list-style: none;
    margin-bottom: 0;
    overflow: hidden;

    ${({ isSubtree, horizontalOverflow }) => css`
        display: ${isSubtree ? 'block' : 'inline-block'};
        padding: 0 0 0 ${isSubtree ? '36px' : '29px'};
        ${!isSubtree &&
            horizontalOverflow &&
            css`
                white-space: nowrap;
            `};
        font-size: ${isSubtree ? '16px' : '15px'};
    `};

    ${({ selection }) => selection && 'padding-left: 0;'}
    ${({ listStyle, ...restProps }) => applyEmotionStyle(listStyle, restProps)};
    ${({ toggleStyles }) => toggleStyles};
`;

export { StyledTreeList };
