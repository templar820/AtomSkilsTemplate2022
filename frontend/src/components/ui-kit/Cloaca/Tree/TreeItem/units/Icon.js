import React from 'react';
import { IconContainer, IconPositioner, DefaultLoadingIcon, DefaultIcon } from './units';

function fitByIconType(icon) {
    return typeof icon === 'function' ? icon : () => icon;
}

function pickProperIcon(isLoading, icon, loadingIcon) {
    return isLoading ? fitByIconType(loadingIcon || DefaultLoadingIcon) : fitByIconType(icon || DefaultIcon);
}

function Icon({ icon, loadingIcon, isLoading, hasSubtree, expanded, handleExpanding, id, ...restProps }) {
    const IconForRendering = pickProperIcon(isLoading, icon, loadingIcon);
    return (
        <IconContainer>
            <IconPositioner onClick={() => hasSubtree && handleExpanding && handleExpanding({ [id]: !expanded })}>
                <IconForRendering {...{ hasSubtree, expanded, ...restProps }} />
            </IconPositioner>
        </IconContainer>
    );
}

export default Icon;
