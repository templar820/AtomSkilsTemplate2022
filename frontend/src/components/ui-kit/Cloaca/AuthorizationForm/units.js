import { css } from '@emotion/core';
import styled from '@emotion/styled';
import Icon from 'lib-ui/InlineIcons';
import applyEmotionStyle from 'lib-root/utils/applyEmotionStyle';

const Header = styled.h2`
    ${({ headerStyle, ...restProps }) => applyEmotionStyle(headerStyle, restProps)};
`;

const IsBlockedWrapper = styled.div`
    ${({ isBlockedStyle, ...restProps }) => applyEmotionStyle(isBlockedStyle, restProps)};
`;

const fieldStyle = (additionalStyle = null) => css`
    max-width: 100%;
    ${additionalStyle};
`;

const buttonStyle = (additionalStyle = null) => css`
    width: 100%;
    ${additionalStyle};
`;

const StyledIcon = styled(Icon)`
    cursor: pointer;
`;

export { Header, IsBlockedWrapper, fieldStyle, buttonStyle, StyledIcon };
