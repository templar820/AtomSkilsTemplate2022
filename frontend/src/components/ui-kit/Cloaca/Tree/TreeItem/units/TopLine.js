import React from 'react';

import TextAndCheckbox from './TextAndCheckbox';
import Icon from './Icon';
import { StyledTopLine } from './units';

const TopLine = ({
    handleExpanding,
    topLineStyle,
    onClick,
    selection,
    parentChecked,
    title,
    onHover,
    nestingLevel,
    nativeTitle,
    ...restProps
}) => (
    <StyledTopLine
        onMouseEnter={() => onHover(true)}
        onMouseLeave={() => onHover(false)}
        {...(nativeTitle ? { title: nativeTitle } : {})}
        {...{ nestingLevel, topLineStyle, selection, parentChecked, onClick, ...restProps }}>
        <Icon {...{ handleExpanding, selection, ...restProps }} />
        <TextAndCheckbox {...{ selection, title, ...restProps }} />
    </StyledTopLine>
);

export default TopLine;
