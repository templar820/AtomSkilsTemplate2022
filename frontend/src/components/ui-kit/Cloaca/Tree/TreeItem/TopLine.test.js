import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import TopLine from './units/TopLine';
import TextAndCheckbox from './units/TextAndCheckbox';
import Icon from './units/Icon';
import { StyledTopLine } from './units/units';

Enzyme.configure({ adapter: new Adapter() });

const onClickFn = jest.fn();
const onHoverFn = jest.fn();

const props = {
    handleExpanding: () => {},
    topLineStyles: { margin: '10px' },
    selection: true,
    parentChecked: true,
    title: 'title',
    nestingLevel: 1,
    onClick: onClickFn,
    onHover: onHoverFn
};

// TODO: need to write all tests.
describe('<TopLine />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<TopLine {...props} />);
    });

    afterEach(() => {
        wrapper.unmount();
    });

    // test must be first, otherwise the snapshot size will be too large
    test('should match snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<TopLine {...props} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('should render correctly', () => {
        const icon = wrapper.find(Icon);
        expect(icon.prop('handleExpanding')).toEqual(props.handleExpanding);
        const styled = wrapper.find(StyledTopLine);
        expect(styled.prop('topLineStyles')).toEqual(props.topLineStyles);
        expect(styled.prop('selection')).toEqual(props.selection);
        expect(styled.prop('parentChecked')).toEqual(props.parentChecked);
        const textAndCheckbox = wrapper.find(TextAndCheckbox);
        expect(textAndCheckbox.prop('title')).toEqual(props.title);
        expect(styled.prop('nestingLevel')).toEqual(props.nestingLevel);
    });

    test('should call onClick method', () => {
        wrapper.simulate('click');
        expect(onClickFn).toHaveBeenCalled();
    });

    test('should call onHover method', () => {
        wrapper.simulate('mouseEnter');
        expect(onHoverFn).toHaveBeenCalledWith(true);
        wrapper.simulate('mouseLeave');
        expect(onHoverFn).toHaveBeenLastCalledWith(false);
    });
});
