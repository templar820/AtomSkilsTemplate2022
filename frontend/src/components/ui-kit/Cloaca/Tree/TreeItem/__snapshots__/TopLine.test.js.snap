// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopLine /> should match snapshot 1`] = `
<StyledTopLine
  nestingLevel={1}
  onClick={[MockFunction]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  parentChecked={true}
  selection={true}
  topLineStyles={
    Object {
      "margin": "10px",
    }
  }
>
  <Icon
    handleExpanding={[Function]}
    selection={true}
    topLineStyles={
      Object {
        "margin": "10px",
      }
    }
  />
  <TextAndCheckbox
    selection={true}
    title="title"
    topLineStyles={
      Object {
        "margin": "10px",
      }
    }
  />
</StyledTopLine>
`;
